org.uqbar.project.wollok.tests/src/org/uqbar/project/wollok/tests/xpect/nonBooleanValueInIfCondition.wlk.xt

```diff
@@ -39,6 +39,12 @@ object p {
 		// XPECT errors --> "Expecting a boolean" at "{ a => a.toString() }"			
 		if ({ a => a.toString() })	 
 			throw new Exception("asd")	
+
+		var a = 2
+		const b = 2
+		// XPECT errors --> "Expecting a boolean" at "a = b"
```

Comment:
What if we produce a more specific validation for this case, so that we can address the specific error of using `=` when you want `==`?

Reply:
![image](https://user-images.githubusercontent.com/4549002/46986084-e5f46780-d0c3-11e8-8904-f552fa7cbadc.png)